### ISSUE TITLE
Error running GPR/DNN

### ISSUE THREAD

**Issue by dvanaken** [cmu-db/ottertune#299]
When I run GPR/DNN I get the following error:
```
12-11-19 01:33:24 [fabfile:1826] INFO : 
-=--------------------------------------------------------------------------------
 [iter:1/5] Downloading the next config from the webserver...
-=--------------------------------------------------------------------------------

12-11-19 01:33:24 [fabfile:1385] DEBUG: Response:

STARTED: Result not ready

12-11-19 01:33:29 [fabfile:1385] DEBUG: Response:

Celery failed to get the next configuration.
STATUS: FAILURE
RESULT ID: 32
TRACEBACK: Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/Users/dva/git/ottertune-fork/server/website/website/tasks/async_tasks.py", line 472, in configuration_recommendation
    X_columnlabels, target_data['X_columnlabels'])
Exception: ('The workload and target data should have identical X columnlabels (sorted knob names)', array(['global._pga_max_size', 'global._smm_max_size',
       'global.db_32k_cache_size', 'global._smm_px_max_size',
       'global.db_cache_size', 'global.db_file_multiblock_read_count',
       'global.log_buffer', 'global.optimizer_adaptive_plans',
       'global.optimizer_features_enable', 'global.shared_pool_size'],
      dtype='<U36'), array(['global._pga_max_size', 'global._smm_max_size',
       'global._smm_px_max_size', 'global.db_32k_cache_size',
       'global.db_cache_size', 'global.db_file_multiblock_read_count',
       'global.log_buffer', 'global.optimizer_adaptive_plans',
       'global.optimizer_features_enable', 'global.shared_pool_size'],
      dtype='<U36'))


12-11-19 01:33:29 [fabfile:1389] ERROR: 
Failed to download the next config: (elapsed: 5s)

Celery failed to get the next configuration.
STATUS: FAILURE
RESULT ID: 32
TRACEBACK: Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/Users/dva/git/ottertune-fork/server/website/website/tasks/async_tasks.py", line 472, in configuration_recommendation
    X_columnlabels, target_data['X_columnlabels'])
Exception: ('The workload and target data should have identical X columnlabels (sorted knob names)', array(['global._pga_max_size', 'global._smm_max_size',
       'global.db_32k_cache_size', 'global._smm_px_max_size',
       'global.db_cache_size', 'global.db_file_multiblock_read_count',
       'global.log_buffer', 'global.optimizer_adaptive_plans',
       'global.optimizer_features_enable', 'global.shared_pool_size'],
      dtype='<U36'), array(['global._pga_max_size', 'global._smm_max_size',
       'global._smm_px_max_size', 'global.db_32k_cache_size',
       'global.db_cache_size', 'global.db_file_multiblock_read_count',
       'global.log_buffer', 'global.optimizer_adaptive_plans',
       'global.optimizer_features_enable', 'global.shared_pool_size'],
      dtype='<U36'))

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/fabric/main.py", line 763, in main
    *args, **kwargs
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/fabric/tasks.py", line 427, in execute
    results['<local-only>'] = task.run(*args, **new_kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/fabric/tasks.py", line 174, in run
    return self.wrapped(*args, **kwargs)
  File "/Users/dva/git/ottertune-driver-internal/fabfile.py", line 1622, in run_experiment
    execute(run_loops, max_iter=max_iter, exp_info=exp_info)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/fabric/tasks.py", line 427, in execute
    results['<local-only>'] = task.run(*args, **new_kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/fabric/tasks.py", line 174, in run
    return self.wrapped(*args, **kwargs)
  File "/Users/dva/git/ottertune-driver-internal/fabfile.py", line 1827, in run_loops
    response = _retval(execute(download_next_config, upload_code=upload_code))
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/fabric/tasks.py", line 427, in execute
    results['<local-only>'] = task.run(*args, **new_kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/fabric/tasks.py", line 174, in run
    return self.wrapped(*args, **kwargs)
  File "/Users/dva/git/ottertune-driver-internal/fabfile.py", line 1390, in download_next_config
    raise Exception(err_msg)
Exception: 
Failed to download the next config: (elapsed: 5s)

Celery failed to get the next configuration.
STATUS: FAILURE
RESULT ID: 32
TRACEBACK: Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/Users/dva/git/ottertune-fork/server/website/website/tasks/async_tasks.py", line 472, in configuration_recommendation
    X_columnlabels, target_data['X_columnlabels'])
Exception: ('The workload and target data should have identical X columnlabels (sorted knob names)', array(['global._pga_max_size', 'global._smm_max_size',
       'global.db_32k_cache_size', 'global._smm_px_max_size',
       'global.db_cache_size', 'global.db_file_multiblock_read_count',
       'global.log_buffer', 'global.optimizer_adaptive_plans',
       'global.optimizer_features_enable', 'global.shared_pool_size'],
      dtype='<U36'), array(['global._pga_max_size', 'global._smm_max_size',
       'global._smm_px_max_size', 'global.db_32k_cache_size',
       'global.db_cache_size', 'global.db_file_multiblock_read_count',
       'global.log_buffer', 'global.optimizer_adaptive_plans',
       'global.optimizer_features_enable', 'global.shared_pool_size'],
      dtype='<U36'))
```

@yangdsh Could you take a look at this when you get a chance?

**Comment by bohanjason** [comment]
@dvanaken  I am working on this. I will create a PR tonight.

**Comment by yangdsh** [comment]
Thanks! @bohanjason 
I also run into this issue on Monday. After I avoided changing the set of knobs to tune, I did not have this issue anymore.

**Comment by bohanjason** [comment]
Fixed in this PR  #300 . Please let me know whether it solves your issues.

**Comment by dvanaken** [comment]
Thank you @bohanjason !

**Comment by dvanaken** [comment]
Fixed in #300
